Userland/Libraries/LibCore/Process.cpp

```diff
@@ -57,64 +56,105 @@ struct ArgvList {
             m_argv.append(nullptr);
         return m_argv;
     }
+};
 
-    void set_working_directory(DeprecatedString const& working_directory)
-    {
-        m_working_directory = working_directory;
-    }
+ErrorOr<Process> Process::spawn(ProcessSpawnOptions const& options)
+{
+#define CHECK(invocation)                  \
+    if (int returned_errno = (invocation)) \
+        return Error::from_errno(returned_errno);
 
-    ErrorOr<pid_t> spawn(Process::KeepAsChild keep_as_child)
-    {
+    posix_spawn_file_actions_t spawn_actions;
+    CHECK(posix_spawn_file_actions_init(&spawn_actions));
+    ScopeGuard cleanup_spawn_actions = [&] {
+        posix_spawn_file_actions_destroy(&spawn_actions);
+    };
+
+    if (options.working_directory.has_value()) {
 #ifdef AK_OS_SERENITY
-        posix_spawn_file_actions_t spawn_actions;
-        posix_spawn_file_actions_init(&spawn_actions);
-        ScopeGuard cleanup_spawn_actions = [&] {
-            posix_spawn_file_actions_destroy(&spawn_actions);
-        };
-        if (!m_working_directory.is_empty())
-            posix_spawn_file_actions_addchdir(&spawn_actions, m_working_directory.characters());
-
-        auto pid = TRY(System::posix_spawn(m_path.view(), &spawn_actions, nullptr, const_cast<char**>(get().data()), System::environment()));
-        if (keep_as_child == Process::KeepAsChild::No)
-            TRY(System::disown(pid));
+        CHECK(posix_spawn_file_actions_addchdir(&spawn_actions, options.working_directory->characters()));
 #else
-        auto pid = TRY(System::posix_spawn(m_path.view(), nullptr, nullptr, const_cast<char**>(get().data()), System::environment()));
-        // FIXME: Support keep_as_child outside Serenity.
-        (void)keep_as_child;
+        TODO();
```

Comment:
Feels like it could do with a comment of what we actually want to do here. Something like `// FIXME: Support chdir file options on platforms that support it`